retrieval/targetmanager.go

```diff
@@ -143,6 +143,25 @@ func (tm *TargetManager) reload() {
 	}
 }
 
+// DroppedTargets returns the targets that were dropped during relabelling.
+func (tm *TargetManager) DroppedTargets() []*Target {
+	tm.mtx.RLock()
+	defer tm.mtx.RUnlock()
+
+	droppedTargets := []*Target{}
+	for _, ps := range tm.targetSets {
+		ps.sp.mtx.RLock()
+
+		for _, t := range ps.sp.droppedTargets {
```

Comment:
This can be simplified by `droppedTargets = append(droppedTargets, ps.sp.droppedTargets...)`